datamapper/repo.py

```diff
@@ -219,6 +229,19 @@ def __init__(self, database: Database):
             await self.__preload(preloaded, subpreloads)
 
 
+def cast_changeset(
+    model_or_changeset: Union[Model, Type[Model], Changeset], values: dict
+) -> Changeset:
+    if isinstance(model_or_changeset, Model):
+        return Changeset(model_or_changeset).cast(values, list(values.keys()))
+    elif isinstance(model_or_changeset, type(Model)):
```

Comment:
This scenario should never happen. There are only two usages of `insert`:

```python
await repo.insert(User(name="Ray"))
await repo.insert(Changeset(User()).cast({"name": "Ray"}, ["name"]))
```

There should only be one usage of `update`:

```python
user = await repo.insert(User(name="Ray"))
await repo.insert(Changeset(user).cast({"name": "Ray"}, ["name"]))
```